code/modules/vending/vending.dm

```diff
@@ -3372,13 +3374,41 @@ ABSTRACT_TYPE(/obj/machinery/vending/jobclothing)
 	New()
 		..()
 		START_TRACKING
+		src.fill_image = image(src.icon, "medchem-0", -1)
+		src.fill_image.plane = PLANE_ABOVE_LIGHTING
+		src.UpdateIcon()
 
 	disposing()
 		. = ..()
 		STOP_TRACKING
 
+	update_icon()
+		..()
+		if (status & (BROKEN|NOPOWER))
+			src.fill_image.icon_state = null
+			src.UpdateOverlays(src.fill_image, "fill_image")
+			return
+		var/total_reagents = 0
+		var/fluid_state = 0
+		for(var/datum/data/vending_product/player_product/R in src.player_list)
+			for(var/obj/item/product in R.contents)
+				if(istype(product, /obj/item/item_box/medical_patches) || istype(product, /obj/item/chem_pill_bottle))
+					for(var/obj/item/reagent_containers/container in product)
+						total_reagents += container.reagents.total_volume
+				if(istype(product, /obj/item/reagent_containers))
+					var/obj/item/reagent_containers/container = product
+					total_reagents += container.reagents.total_volume
+
+		if(!total_reagents) //only show it as empty if it truly has Nothing at All, chemically
+			fluid_state = 0
+		else
+			fluid_state = round(clamp((total_reagents / 500 * 9 + 1), 1, 9)) //it'll jump up a state every ~55 reagents added or so
+		src.fill_image.icon_state = "medchem-[fluid_state]"
+		src.UpdateOverlays(src.fill_image, "fill_image")
```

Comment:
Since you're never swapping out the overlay you should just be able to call `UpdateOverlays` once and then change the icon with your image reference?

Reply:
unless I'm misunderstanding the suggestion trying this has made the icon not actually swap in-game? I *think* most overlays use UpdateOverlays to update icon changes like this, at least, though again it's entirely possible I just don't Get What You Mean :sheltersweat: